applications/StructuralMechanicsApplication/custom_elements/membrane_element.hpp

```diff
@@ -0,0 +1,415 @@
+      int Step = 0) override;
+
+    void GetSecondDerivativesVector(
+      Vector& rValues,
+      int Step = 0) override;
+
+    int Check(const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateOnIntegrationPoints(
+      const Variable<array_1d<double, 3>>& rVariable,
+      std::vector<array_1d<double, 3>>& rOutput,
+      const ProcessInfo& rCurrentProcessInfo) override;
+
+    void GetValueOnIntegrationPoints(
+      const Variable<array_1d<double, 3>>& rVariable,
+      std::vector<array_1d<double, 3>>& rOutput,
+      const ProcessInfo& rCurrentProcessInfo) override;
+
+
+      /**
+     * @brief Creates a given local coordinate system
+     * @param rBaseVectors the new local coordinate system
+     * @param rLocalBaseVectors the base coordinate system
+     */
+    void TransformBaseVectors(array_1d<Vector,2>& rBaseVectors,
+     const array_1d<Vector,2>& rLocalBaseVectors);
+
+
+      /**
+     * @brief Creates the transformation matrix for a given local coordinate system
+     * @param rTransformationMatrix the transformation matrix
+     * @param rTransformedBaseVectors local coordinate system
+     * @param rLocalReferenceBaseVectors the base coordinate system
+     */
+    template <class T>
+    void InPlaneTransformationMatrix(Matrix& rTransformationMatrix, const array_1d<Vector,2>& rTransformedBaseVectors,
+      const T& rLocalReferenceBaseVectors);
+
+    void CalculateMassMatrix(MatrixType& rMassMatrix,ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLumpedMassVector(VectorType& rMassVector);
+
+    void AddExplicitContribution(
+      const VectorType& rRHSVector, const Variable<VectorType>& rRHSVariable,
+      Variable<array_1d<double, 3>>& rDestinationVariable,
+      const ProcessInfo& rCurrentProcessInfo) override;
+
+    void AddExplicitContribution(
+      const VectorType& rRHSVector,
+      const Variable<VectorType>& rRHSVariable,
+      Variable<double >& rDestinationVariable,
+      const ProcessInfo& rCurrentProcessInfo) override;
+
+    void Calculate(const Variable<Matrix>& rVariable,
+      Matrix& rOutput, const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateAndAddBodyForce(VectorType& rRightHandSideVector);
+
+    void CalculateDampingMatrix(MatrixType& rDampingMatrix, ProcessInfo& rCurrentProcessInfo) override;
+
+
+      /**
+     * @brief Calculates the principle vectors
+     * @param rPrincipleVector the principle vectors
+     * @param rNonPrincipleVector reference state
+     */
+    void PrincipleVector(Vector& rPrincipleVector, const Vector& rNonPrincipleVector);
+
+
+      /**
+     * @brief Checks for taunt/slack/wrinkles
+     * @param rWrinklingStateArray the current wrinkling state
+     * @param rStress the stress
+     * @param rStrain the strain
+     */
+    void CheckWrinklingState(array_1d<bool,3>& rWrinklingStateArray, const Vector& rStress, const Vector& rStrain);
+
+protected:
+  ConstitutiveLaw::Pointer mpConstitutiveLaw = nullptr;
```

Comment:
I think you should go for a vector. Talking about historic CLs.

Reply:
I will follow @philbucher and get rid of the claw member